wcfsetup/install/files/lib/util/StringUtil.class.php

```diff
@@ -200,51 +186,26 @@ public static function formatInteger($integer): string
 
     /**
      * Formats a double.
-     *
-     * @param double $double
-     * @param int $maxDecimals
      */
-    public static function formatDouble($double, $maxDecimals = 0): string
+    public static function formatDouble(float $double, int $maxDecimals = 0): string
     {
-        // round
-        $double = (string)\round($double, ($maxDecimals > 0 ? $maxDecimals : 2));
+        $double = \round($double, ($maxDecimals > 0 ? $maxDecimals : 2));
 
-        // consider as integer, if no decimal places found
-        if (!$maxDecimals && \preg_match('~^(-?\d+)(?:\.(?:0*|00[0-4]\d*))?$~', $double, $match)) {
-            return self::formatInteger($match[1]);
-        }
+        $double = self::getNumberFormatter()->format($double);
 
-        // remove last 0
-        if ($maxDecimals < 2 && \substr($double, -1) == '0') {
-            $double = \substr($double, 0, -1);
+        if ($double < 0) {
+            $double = self::formatNegative($double);
         }
 
-        // replace decimal point
-        $double = \str_replace('.', WCF::getLanguage()->get('wcf.global.decimalPoint'), $double);
-
-        // add thousands separator
-        $double = self::addThousandsSeparator($double);
-
-        // format minus
-        return self::formatNegative($double);
+        return $double;
     }
 
     /**
      * Adds thousands separators to a given number.
-     *
-     * @param mixed $number
      */
-    public static function addThousandsSeparator($number): string
+    public static function addThousandsSeparator(int|float $number): string
```

Comment:
Should the majority of the functions be deprecated and perhaps just `formatNumeric` remain?